File: 5_claude_sol1.cpp
#include <bits/stdc++.h>
using namespace std;

const long long MOD = 998244353;

int main() {
    int T;
    scanf("%d", &T);
    while (T--) {
        int n;
        scanf("%d", &n);
        vector<string> a(n);
        for (int i = 0; i < n; i++) {
            int x;
            scanf("%d", &x);
            a[i] = to_string(x);
        }
        sort(a.begin(), a.end(), [](const string &x, const string &y) {
            return x + y > y + x;
        });
        long long res = 0;
        for (int i = 0; i < n; i++) {
            long long p10 = 1;
            for (size_t j = 0; j < a[i].size(); j++) p10 = (p10 * 10) % MOD;
            res = (res * p10 + (stoll(a[i]) % MOD)) % MOD;
        }
        printf("%lld\n", res);
    }
    return 0;
}

File: 5_claude_sol1.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        final long MOD = 998244353;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer in = new StreamTokenizer(br);
        in.nextToken();
        int t = (int) in.nval;
        StringBuilder sb = new StringBuilder();
        while (t-- > 0) {
            in.nextToken();
            int n = (int) in.nval;
            String[] a = new String[n];
            for (int i = 0; i < n; i++) {
                in.nextToken();
                a[i] = String.valueOf((long) in.nval);
            }
            Arrays.sort(a, (x, y) -> (y + x).compareTo(x + y));
            long res = 0;
            for (String s : a) {
                long p10 = 1;
                for (int j = 0; j < s.length(); j++) p10 = (p10 * 10) % MOD;
                res = (res * p10 + Long.parseLong(s) % MOD) % MOD;
            }
            sb.append(res).append('\n');
        }
        System.out.print(sb);
    }
}
